web/src/date-utils.ts

```diff
@@ -66,7 +67,14 @@ export const dateToFormat = (date: Date | number, format: DateFormat): string =>
       case DateFormat.TimeFull: {
         const fractionalSeconds =
           typeof date === 'number' ? Math.floor(date % 1000) : date.getMilliseconds();
```

Comment:
I think there is a bug here. For a time like `12:00:00.001`, `getMilliseconds` will return `1`, so you will end up returning `12:00:00.1`.

Reply:
Added some unit tests so we check this cases

Reply:
💯